godel_scan_analysis/src/scan_roughness_scoring.cpp

```diff
@@ -0,0 +1,136 @@
+#include "godel_scan_analysis/scan_roughness_scoring.h"
+
+#include "godel_scan_analysis/scan_algorithms.h"
+
+#include <ros/ros.h> // debug
+
+#include <math.h> // isfinite
+
+/*
+  I have a variety of ideas for making this code faster if it was ever needed:
+  1) Keep scan data as an array of x and array of y for better cache coherency on windowing
+  2) The windowing could be made to operate in O(n) - touching every point only once
+  3) Pre-calculate x values (which are known)
+  4) Pre-calculate colors
+  5) Perhaps return an index map like PCL does from the filter function so
+     we don't need to make a copy of the scan
+  6) Pre-allocate or use stack space (std::array?) for storage space
+*/
+
+const static double DEFAULT_MAX_SCORE = 0.00003;  // RMS value about a point that is considered out of spec
+const static double DEFAULT_MIN_SCORE = 0.0;      // Min RMS value; Used for coloring together with above
+
+namespace
+{
+  typedef godel_scan_analysis::RoughnessScorer::Cloud Cloud;
+  typedef godel_scan_analysis::RoughnessScorer::ColorCloud ColorCloud;
+
+  const static unsigned WINDOW_SIZE = 30;
```

Comment:
Since the namespace has not effect on this variable perhaps it be best to declare it above with the rest of the other global constants for the sake of clarity?